helper/chksum.c

```diff
@@ -206,10 +204,14 @@ static inline int odph_process_l3_hdr(odp_packet_t odp_pkt,
 			ipv4_hdr_ptr = &ipv4_hdr;
 		}
 
-		addrs_ptr = (uint16_t *)(void *)&ipv4_hdr_ptr->src_addr;
-		addrs_len = 2 * ODPH_IPV4ADDR_LEN;
 		protocol  = ipv4_hdr_ptr->proto;
 		l3_len    = odp_be_to_cpu_16(ipv4_hdr_ptr->tot_len);
+
+		/* calculate IP pseudo header src and dest addr checksum */
+		sum       = ipv4_hdr_ptr->src_addr & 0xFFFF;
+		sum      += ipv4_hdr_ptr->src_addr >> 16 & 0xFFFF;
```

Comment:
The "& 0xFFFF" part is unnecessary here. And the same applies to other occurences of this pattern.

Reply:
It's more of the code readability.. I believe its done in other places in code too. But I have removed unnecessary masking